code/clue.js

```diff
@@ -1,59 +1,267 @@
+  age: 'Age: ' + 29,
+  image: 'assets/plum.png',
+  occupation: 'Occupation: biologist',
+  favouriteWeapon: 'poison'
+}
+
+const msScarlet = {
+  firstName: 'Cassandra',
+  lastName: 'Scarlet',
+  color: 'pink',
+  description: 'She is super-smart.',
+  age: 'Age: ' + 26,
+  image: 'assets/scarlet.png',
+  occupation: 'Occupation: journalist',
+  favouriteWeapon: 'bat'
+}
+
+const mrsWhite = {
+  firstName: 'Evie',
+  lastName: 'White',
+  color: 'white',
+  description: 'She is no old biddy.',
+  age: 'Age: ' + 62,
+  image: 'assets/white.png',
+  occupation: 'Occupation: housekeeper',
+  favouriteWeapon: 'axe'
+}
+
+// Objects for weapons
 
 const rope = {
   name: 'rope',
   weight: 10
 }
 
-// THE ROOMS ONLY HAS A NAME SO NO NEED FOR OBJECTS THERE.
+const knife = {
+  name: 'knife',
+  id: 'knife',
+  weight: 5
+}
+
+const candlestick = {
+  name: 'candlestick',
+  weight: 25
+}
+
+const dumbbell = {
+  name: 'dumbbell',
+  id: 'dumbbell',
+  weight: 30
+}
+
+const poison = {
+  name: 'poison',
+  id: 'poison',
+  weight: 5
+}
+
+const axe = {
+  name: 'axe',
+  id: 'axe',
+  weight: 25
+}
+
+const bat = {
+  name: 'bat',
+  id: 'bat',
+  weight: 20
+}
+
+const trophy = {
+  name: 'trophy',
+  weight: 30
+}
+
+const pistol = {
+  name: 'pistol',
+  weight: 25
+}
+
 
-// NOW GROUP ALL SUSPECTS, WEAPONS AND ROOMS IN ARRAYS LIKE THIS:
+// Arrays for suspects, weapons and rooms
 
 const suspects = [
   mrGreen,
-  mrsWhite
-  // ...  and the rest
+  mrsWhite,
+  mrPlum,
+  msScarlet,
+  mrsPeacock,
+  colonelMustard
 ]
 
-const weapons = []
+const weapons = [
+  rope,
+  knife,
+  candlestick,
+  dumbbell,
+  poison,
+  axe,
+  bat,
+  trophy,
+  pistol
+]
+
+const rooms = [
+  'diningroom', 
+  'conservatory', 
+  'kitchen', 
+  'study', 
+  'library', 
+  'billiardroom', 
+  'lounge', 
+  'ballroom', 
+  'hall', 
+  'spa', 
+  'livingroom', 
+  'observatory', 
+  'theatre', 
+  'guesthouse', 
+  'patio'
+]
 
-const rooms = []
+// Selects randomly one item from the array passed into the function
 
-// THIS FUNCTION WILL RANDOMLY SELECT ONE ITEM FROM THE ARRAY THAT YOU PASS IN TO THE FUNCTION.
-// YOU DON'T NEED TO CHANGE THIS, JUST TRY TO UNDERSTAND IT. AND HOW TO USE IT.
 const randomSelector = array => {
   return array[Math.floor(Math.random() * array.length)]
-}
+};
 
-// CREATE AN OBJECT THAT KEEPS THE MYSTERY.
-// With a killer, a weapon and a room.
+// Creates an object that stores the mystery 
 // The values will be set later.
 
-// This function will be invoked when you click on the killer card.
+const mystery = {
+  killer: undefined,
+  weapon: undefined,
+  room: undefined,
+};
+
+// FUNCTION - this function will run when you click on the killer card.
+
 const pickKiller = () => {
-  // This will randomly select a killer from the suspects. And add that to the mystery object.
-  mystery.killer = randomSelector(suspects)
+  mystery.killer = randomSelector(suspects);
+  document.getElementById('loader1').style.display = 'block';
+
```

Comment:
is here needed an empty line?(: 
anyway it's a very good thing about the setTimeout!